src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -118,6 +118,29 @@ namespace winrt::TerminalApp::implementation
         _systemRowsToScroll = _ReadSystemRowsToScroll();
     }
 
+    bool TerminalPage::IsElevated() const noexcept
+    {
+        // use C++11 magic statics to make sure we only do this once.
+        // This won't change over the lifetime of the application
+
+        static const bool isElevated = []() {
+            // *** THIS IS A SINGLETON ***
+            auto result = false;
+
+            // GH#2455 - Make sure to try/catch calls to Application::Current,
+            // because that _won't_ be an instance of TerminalApp::App in the
+            // LocalTests
+            try
```

Comment:
How does this try-catch work with the magic static? If the try fails, is `isElevated` permanently stuck as `false`?

(Also, is this even a valid concern? Does Application::Current fail often?)

Reply:
> does Application::Current fail often

literally only in the tests, because Application::Current isn't a TerminalApp::App in the tests. With the magic static, we catch the exception as it's getting initialized, return false, and then put false in the static member to store forever

Reply:
Right now AppLogic just sets a member on itself already when it is created, but also the implementation of `IsUserAdmin` doesn't actually rely on any part of AppLogic's state. Would it make sense to then make a free function/static function `IsElevated` somewhere else that could just be used directly without multiple layers of caching and indirection? As an optimization that `IsElevated` function could store a static member, should you choose to.